### PR TITLE
v2-api: Grpc Json transcoder

### PR DESCRIPTION
**Pull request by rshriram**
Signed-off-by: Shriram Rajagopalan <shriram@us.ibm.com>
*title*: v2-api : Grpc Json transcoder

*Description*: Migrating Grpc Json transcoder to v2-api and some more cleanups
*Risk Level*: Low
*Testing*: existing unit and integration tests

### CODE REVIEW

File: source/common/config/filter_json.cc [hunk 1]
```diff
@@ -289,6 +289,30 @@ void FilterJson::translateHealthCheckFilter(
...
+  JSON_UTIL_SET_STRING(json_config, proto_config, proto_descriptor);
+  auto* services = proto_config.mutable_services();
+  for (const auto& service_name : json_config.getStringArray("services")) {
+    *services->Add() = service_name;
+  }
+
+  if (json_config.hasObject("print_options")) {
+    auto json_print_options = json_config.getObject("print_options");
+    auto* proto_print_options = proto_config.mutable_print_options();
+    proto_print_options->set_add_whitespace(
```
Comment (mattklein123) [inline source/common/config/filter_json.cc]: JSON_UTIL_SET_BOOL for all of these?

Comment (rshriram) [reply source/common/config/filter_json.cc]: I thought it only worked on wrapped types... let me try

Comment (rshriram) [reply source/common/config/filter_json.cc]: Nope.. it won't work.. the macro expects wrapped types..

Comment (mattklein123) [reply source/common/config/filter_json.cc]: OK no problem. Thanks for trying.

**Pull request merged**